lib/streamlit/dataframe_util.py

```diff
@@ -176,23 +193,28 @@ def is_unevaluated_data_object(obj: object) -> bool:
         or is_pyspark_data_object(obj)
         or is_snowpandas_data_object(obj)
         or is_modin_data_object(obj)
+        or inspect.isgeneratorfunction(obj)
     )
 
 
+def is_pandas_data_object(obj: object) -> bool:
```

Comment:
I think it'd be best if we had unit tests also for these helper functions to narrow down issues quickly 🙂

Reply:
Added 👍